app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java

```diff
@@ -221,4 +242,32 @@ private void getDeviceLocation() {
             });
         }
     }
+
+    private void setPinnieCount() {
+        FirebaseDriver driver = FirebaseDriver.getInstance();
+        String uid = driver.getCurrentUser().getUid();
+
+        if(driver.getCachedPinnies() != null){
+            pinnieCount = driver.getCachedPinnies();
+            setPinniesUI();
+            return;
+        }
+
+        driver.getPinnies(uid).addOnCompleteListener(task -> {
```

Comment:
I originally implemented `getPinnies` this way since I figured the flexibility couldn't hurt, i.e. this driver function is a generic way to access the `private` firestore collection.